pilot/pkg/networking/util/util.go

```diff
@@ -277,21 +301,35 @@ func IsTCPMetadataExchangeEnabled(node *model.Proxy) bool {
 }
 
 // ConvertLocality converts '/' separated locality string to Locality struct.
-func ConvertLocality(locality string) *core.Locality {
+func ConvertLocality(locality string) *corev3.Locality {
+	if locality == "" {
+		return &corev3.Locality{}
+	}
+
+	region, zone, subzone := SplitLocality(locality)
+	return &corev3.Locality{
+		Region:  region,
+		Zone:    zone,
+		SubZone: subzone,
+	}
+}
+
+// ConvertLocality converts '/' separated locality string to Locality struct.
+func ConvertLocalityV3(locality string) *corev3.Locality {
 	if locality == "" {
-		return &core.Locality{}
+		return &corev3.Locality{}
 	}
 
 	region, zone, subzone := SplitLocality(locality)
-	return &core.Locality{
+	return &corev3.Locality{
 		Region:  region,
 		Zone:    zone,
 		SubZone: subzone,
 	}
 }
 
 // ConvertLocality converts '/' separated locality string to Locality struct.
-func LocalityToString(l *core.Locality) string {
+func LocalityToString(l *corev3.Locality) string {
```

Comment:
don't we need this for v2? Probably not since you could move this to v3 directly?

Reply:
all locality stuff is v3